packages/fxa-auth-server/test/local/senders/email.js

```diff
@@ -16,216 +16,790 @@ const sinon = require('sinon');
-  'verificationReminderSecondEmail',
-  'verifyEmail',
-  'verifyLoginEmail',
-  'verifyLoginCodeEmail',
-  'verifyPrimaryEmail',
-  'verifySecondaryEmail',
-  'passwordResetAccountRecoveryEmail',
-  'postAddAccountRecoveryEmail',
-  'postRemoveAccountRecoveryEmail',
-];
-
-const typesContainSupportLinks = new Set([
-  'lowRecoveryCodesEmail',
-  'newDeviceLoginEmail',
-  'passwordChangedEmail',
-  'passwordResetEmail',
-  'postAddTwoStepAuthenticationEmail',
-  'postChangePrimaryEmail',
-  'postRemoveSecondaryEmail',
-  'postRemoveTwoStepAuthenticationEmail',
-  'postVerifyEmail',
-  'recoveryEmail',
-  'verificationReminderFirstEmail',
-  'verificationReminderSecondEmail',
-  'verifyEmail',
-  'verifyLoginCodeEmail',
-  'verifyPrimaryEmail',
-  'verifySecondaryEmail',
-  'passwordResetAccountRecoveryEmail',
-  'postAddAccountRecoveryEmail',
-  'postRemoveAccountRecoveryEmail',
-]);
-
-const typesContainPasswordResetLinks = new Set([
-  'passwordChangedEmail',
-  'passwordResetEmail',
-  'passwordResetRequiredEmail',
+const MESSAGE = {
+  acceptLanguage: 'en',
+  code: 'abc123',
+  deviceId: 'foo',
+  location: {
+    city: 'Mountain View',
+    country: 'USA',
+    stateCode: 'CA',
+  },
+  email: 'a@b.com',
+  flowBeginTime: Date.now(),
+  flowId: '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+  ip: '219.129.234.194',
+  locations: [],
+  numberRemaining: 2,
+  primaryEmail: 'c@d.com',
+  productId: 'wibble',
+  service: 'sync',
+  timeZone: 'America/Los_Angeles',
+  tokenCode: 'abc123',
+  type: 'secondary',
+  uaBrowser: 'Firefox',
+  uaBrowserVersion: '70.0a1',
+  uaOS: 'Windows',
+  uaOSVersion: '10',
+  uid: 'uid',
+  unblockCode: 'AS6334PK',
+};
+
+const COMMON_TESTS = new Map([
+  ['from', { test: 'equal', expected: config.smtp.sender }],
+  ['sender', { test: 'equal', expected: config.smtp.sender }],
+  [
+    'headers',
+    new Map([
+      ['X-Device-Id', { test: 'equal', expected: MESSAGE.deviceId }],
+      ['X-Email-Service', { test: 'equal', expected: 'fxa-auth-server' }],
+      ['X-Flow-Begin-Time', { test: 'equal', expected: MESSAGE.flowBeginTime }],
+      ['X-Flow-Id', { test: 'equal', expected: MESSAGE.flowId }],
+      ['X-Service-Id', { test: 'equal', expected: MESSAGE.service }],
+      [
+        'X-SES-CONFIGURATION-SET',
+        { test: 'equal', expected: config.smtp.sesConfigurationSet },
+      ],
+      ['X-Uid', { test: 'equal', expected: MESSAGE.uid }],
+    ]),
+  ],
 ]);
 
-const typesContainPasswordChangeLinks = new Set([
-  'newDeviceLoginEmail',
-  'verifyLoginEmail',
-  'verifyLoginCodeEmail',
-  'verifyPrimaryEmail',
-  'postAddTwoStepAuthenticationEmail',
-  'postChangePrimaryEmail',
-  'postRemoveTwoStepAuthenticationEmail',
-  'postVerifySecondaryEmail',
-  'postConsumeRecoveryCodeEmail',
-  'postNewRecoveryCodesEmail',
-  'passwordResetAccountRecoveryEmail',
-  'postAddAccountRecoveryEmail',
-  'postRemoveAccountRecoveryEmail',
+// prettier-ignore
+const TESTS = new Map([
```

Comment:
And these are the assertions that are email-specific.

`// prettier-ignore` because I expressly want it to be easy to find a particular key by eye, but that's harder to do when everything is spread out over multiple lines and you have to scroll much further. I'm not bothered by the long line lengths because it's just data and as long as you can find the key that interests you, it's easy to scroll right at the point of interest. If it was code, I'd absolutely have a problem with the line lengths.